File: src/scraper/entities/item.entity.ts
import { Column, Entity, ManyToMany, PrimaryGeneratedColumn } from 'typeorm';
import { WeaponLevelRank } from './weapon_level_rank.entity';

@Entity()
export class ItemEntity {

  @PrimaryGeneratedColumn()
  id: number;

  @Column({ unique: true })
  name: string;

  @Column()
  subTypes: string;

  @Column()
  description: string;

  @Column()
  rank: number;

  @Column()
  type: string;

  @Column()
  imageUrl: string;

  @ManyToMany(() => WeaponLevelRank, levelRank => levelRank.items)
  levelRanks: WeaponLevelRank[];

}

File: mobile/lib/domains/item/item_entity.dart

import 'package:freezed_annotation/freezed_annotation.dart';

part 'item_entity.freezed.dart';
part 'item_entity.g.dart';

@freezed
class ItemEntity with _$ItemEntity {
  const factory ItemEntity({
    int? id,
    String? name,
    String? subTypes,
    String? description,
    int? rank,
    String? type,
    String? imageUrl,
  }) = _ItemEntity;

  factory ItemEntity.fromJson(Map<String, dynamic> json) =>
      _$ItemEntityFromJson(json);
}